homeassistant/components/mhz19/sensor.py

```diff
@@ -107,9 +117,10 @@ def update(self):
     def extra_state_attributes(self):
         """Return the state attributes."""
         result = {}
-        if self._sensor_type == SENSOR_TEMPERATURE and self._ppm is not None:
+        sensor_type = self.entity_description.key
+        if sensor_type == SENSOR_TEMPERATURE and self._ppm is not None:
             result[ATTR_CO2_CONCENTRATION] = self._ppm
-        if self._sensor_type == SENSOR_CO2 and self._temperature is not None:
+        elif sensor_type == SENSOR_CO2 and self._temperature is not None:
             result[ATTR_TEMPERATURE] = self._temperature
         return result
```

Comment:
This seems a bit redundant, do we really need to include the state of the other sensor as an attribute?